lib/timers.js

```diff
@@ -164,56 +196,82 @@ exports._unrefActive = function(item) {
 // Appends a timer onto the end of an existing timers list, or creates a new
 // TimerWrap backed list if one does not already exist for the specified timeout
 // duration.
-function insert(item, unrefed, start) {
+function insert(item, refed, start) {
   const msecs = item._idleTimeout;
-  if (msecs < 0 || msecs === undefined) return;
-
-  if (typeof start === 'number') {
-    item._idleStart = start;
-  } else {
-    item._idleStart = TimerWrap.now();
-  }
+  if (msecs < 0 || msecs === undefined)
+    return;
 
-  const lists = unrefed === true ? unrefedLists : refedLists;
+  item._idleStart = start;
 
   // Use an existing list if there is one, otherwise we need to make a new one.
   var list = lists[msecs];
   if (list === undefined) {
     debug('no %d list was found in insert, creating a new one', msecs);
-    lists[msecs] = list = new TimersList(msecs, unrefed);
+    lists[msecs] = list = new TimersList(start, msecs);
```

Comment:
If you stent this `expiry` instead you'd have to do a tad less math in the lines that follow (mostly code clarity)